main/src/vtr_radar/src/modules/odometry/odometry_gyro_module.cpp

```diff
@@ -0,0 +1,174 @@
+// Copyright 2021, Autonomous Space Robotics Lab (ASRL)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * \file odometry_gyro_module.cpp
+ * \author Yuchen Wu, Keenan Burnett, Autonomous Space Robotics Lab (ASRL)
+ */
+#include "vtr_radar/modules/odometry/odometry_gyro_module.hpp"
+
+#include "vtr_radar/utils/nanoflann_utils.hpp"
+
+#include "steam/evaluable/p2p/yaw_vel_error_evaluator.hpp"
+
+namespace vtr {
+namespace radar {
+
+using namespace tactic;
+using namespace steam;
+using namespace steam::se3;
+using namespace steam::traj;
+using namespace steam::vspace;
+
+auto OdometryGyroModule::Config::fromROS(const rclcpp::Node::SharedPtr &node,
+                                        const std::string &param_prefix)
+    -> ConstPtr {
+  auto config = std::make_shared<Config>();
+  // clang-format off
+  // motion compensation
+  const auto qcd = node->declare_parameter<std::vector<double>>(param_prefix + ".traj_qc_diag", std::vector<double>());
+  if (qcd.size() != 6) {
+    std::string err{"Qc diagonal malformed. Must be 6 elements!"};
+    CLOG(ERROR, "radar.odometry_icp") << err;
+    throw std::invalid_argument{err};
+  }
+  config->traj_qc_diag << qcd[0], qcd[1], qcd[2], qcd[3], qcd[4], qcd[5];
+  config->gyro_cov = node->declare_parameter<double>(param_prefix + ".gyro_cov", 0.1);
```

Comment:
Better form is to define the default value in the hpp file and use `config->gyro_cov` as the default value. Please change for all config parameters.

Reply:
done